core/modules/admin_bar/admin_bar.inc

```diff
@@ -720,6 +723,57 @@ function admin_bar_links_page() {
   return $links;
 }
 
+/**
+ * Build Locale links.
+ *
+ * @see theme_admin_bar_links()
+ */
+function admin_bar_links_locale() {
+  if (!module_exists('locale') || !language_multilingual()) {
+    return array();
+  }
+
+  global $language;
+  $language_default = $language;
+
+  $links['locale'] = array(
+    '#title' => theme('admin_bar_icon_locale'),
+    '#attributes' => array('class' => array('admin-bar-locale-links')),
+  );
+
+  $path = backdrop_is_front_page() ? '<front>' : $_GET['q'];
+  $language_links = language_negotiation_get_switch_links(LANGUAGE_TYPE_INTERFACE, $path);
+
+  if (!isset($language_links->links)) {
+    return array();
+  }
+
+  foreach ($language_links->links as $language_link) {
+    if ($language_link['language']->langcode == $language_default->langcode) {
+      continue;
+    }
+
+    $links['locale'][$language_link['language']->langcode] = array(
+      '#attributes' => array('class' => array('locale')),
+      '#title' => $language_link['language']->name,
```

Comment:
I suggest changing this to:
```suggestion
      '#title' => $language_link['language']->native,
```

This will match the behavior of the Language Switcher block, where all languages are shown in their native spellings.